dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/OpenAIChatMessageContent.cs

```diff
@@ -110,6 +116,7 @@ public IReadOnlyList<OpenAIFunctionToolCall> GetOpenAIFunctionToolCalls()
             }
 
             // Add the additional entry.
+            newDictionary.Add(FunctionToolCallsProperty, toolCalls.OfType<ChatCompletionsFunctionToolCall>().ToList());
```

Comment:
Are both necessary?

Reply:
ONe don't serialize the function call specific details, the other does.

Reply:
ONe way to resolve this later is to actually transform the ChatCompletionsToolCall object in a JsonElement and add it to the dictionary, or better approach.

Reply:
Serializing both is going to duplicate information. If we have to serialize non-function tool calls, can we filter it down to just the ones that aren't functions? For now that should be empty, so most of the time we don't need to add anything.

Reply:
Or just do the jsonelement thing now, as you suggest

Reply:
Moving to internal

Reply:
I moved the properties to internal and I will be changing this behavior internally.

Looking currently in a definitive solution, I tried the JsonElement approach but it didn't work initially, will look at this in more detail.